src/main/java/org/logstash/dissect/Dissector.java

```diff
@@ -53,34 +56,58 @@ public Dissector(String mapping) {
         Collections.sort(saveableFields, new FieldComparator());
     }
 
-    public int dissect(byte[] source, Map<String, Object> map) {
-        int pos = dissectValues(source);
-        ValueResolver resolver = new ValueResolver(source, fieldValueRefMap);
+    public int dissect(byte[] source, Map<String, Object> keyValueMap) {
+        ResolveValue resolveValue = resolve(source);
 
         for (Field field : saveableFields) {
-            field.append(map, resolver);
+            field.append(keyValueMap, resolveValue.resolver);
```

Comment:
Testing = unit tests

Its only purpose is to mimic an Event (and possibly so that the Ingest folks can see that it will work for their IngestDocument `sourceAndMetadata` HashMap)